### PR TITLE
Prevent sleep timer from running on native platforms

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
This sleep timer thing is wreaking certain havoc on native iOS and Android. Not sure why, but probably it should be disabled for now to restore performance on native apps. My current theory is that other work on the JS thread can cause the timer to get out of whack and cause reconnection callbacks to fire.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/157445

### Tests
**iOS/Android**
1. Test navigating around the app
2. Verify that the reconnection callbacks are not firing randomly
3. Verify that putting the app into the background and bringing it active again does fire the reconnection callbacks

**Web/Desktop**
1. Test navigating around the app
2. Close your laptop screen and reopen it
3. Verify the clock timer works for this scenario
4. Verify that minimizing the browser or app window and maximizing it again does not cause the reconnection callbacks to fire 

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/libs/AppStateMonitor/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+function addBecameActiveListener(callback) {
+    /**
+     * @param {String} state
+     */
+    function appStateChangeCallback(state) {
+        if (
+            shouldReportActivity
+            && (appState === CONST.APP_STATE.INACTIVE || appState === CONST.APP_STATE.BACKGROUND)
+            && state === CONST.APP_STATE.ACTIVE
```
Comment (thienlnam) [inline src/libs/AppStateMonitor/index.js:21]: I think I understand this but basically this is only change the app state if the user has become active and the stored state is not active right?

File: src/libs/AppStateMonitor/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        ) {
+            callback();
+        }
+        appState = state;
+    }
+    AppState.addEventListener('change', appStateChangeCallback);
+    return () => {
+        AppState.removeEventListener('change', appStateChangeCallback);
+    };
```
Comment (thienlnam) [inline src/libs/AppStateMonitor/index.js:30]: Could you explain what is happening here? I'm not totally sure what this is doing, is it initializing with an event listener and then whenever it is returned it gets removed?

**Review by thienlnam — APPROVED** [review]
Code looks good and it works, just had some questions about some of the code

File: src/libs/AppStateMonitor/index.js (hunk 1 above)
Comment (marcaaron) [reply src/libs/AppStateMonitor/index.js:21]: Correct. We set up the state (assume to be active when the app inits) then listen for changes and save it locally each time so we can detect a change from one state to another.

File: src/libs/AppStateMonitor/index.js (hunk 2 above)
Comment (marcaaron) [reply src/libs/AppStateMonitor/index.js:30]: Sure! So, `addBecameActiveListener` is both setting up an event listener and then returning a function that we can call later to remove the listener. The `appStateChangeCallback` exists in the closure created by calling the outermost function. It's just sort of a convenient pattern for cleaning up the listener we created and I borrowed it from how `NetInfo` works.

**Comment by thienlnam** [comment]
Sweet, thanks for the explanations, makes sense!

**Comment by isagoico** [comment]
@marcaaron Any way we could test the callbacks on our side? Or should we just test that the app is updated after reopening it from background?

**Comment by marcaaron** [comment]
>Or should we just test that the app is updated after reopening it from background?

That should work! It's the only way to test this without having to look at the logs to verify that they are running correctly.

**Pull request merged**